### DISCUSSION TITLE
How to achieve conditional rendering of user control children

### DISCUSSION THREAD

**Discussion by devtud** [flet-dev/flet#766] [Q&A] [answered]
Hi,

I have a user control with some conditions in `build()`. Below it is just a POC. I assumed conditional rendering of the children is straight forward and easily achievable with calling `update()` when the data in my user control changes and it needs to be re-rendered.

```python
class MyControl(ft.UserControl):
    def __init__(self):
        self.dynamic_member = 0

    def some_callback(self):
        self.dynamic_member += 1
        self.update()

    def build(self):
        if self.dynamic_member == 0:
            return ft.Text("oops, it's 0")
        return ft.Text(f"it's {self.dynamic_member}")
```

I expect that by calling `self.update()`, the `build()` will be triggered and my user control will be updated according to the new value of `self.dynamic_member`, but this doesn't happen. I get the "oops, it's 0" text whenever `some_callback()` is called (as `build()` is never called again).

There is not much documentation on this area and I suppose I don't have a good understanding on how flet works, so I'm stuck here. Any help (with some explanations if possible) would be much appreciated.

**Comment by Francois-Belanger** [comment, marked as answer]
Hi, you have to work by reference to do this simply.

```
class MyControl(ft.UserControl):
    def __init__(self):
        self.dynamic_member = 0
        self.text_box = ft.Text("oops, it's 0")

    def some_callback(self):
        self.dynamic_member += 1
        self.text_box .value = f"it's {self.dynamic_member}"
        self.update()

    def build(self):
        return self.text_box
```
